LibTest/js_interop/NullableUndefineableJSAnyExtension/isNull_t01.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion bool get isNull
+/// Whether this value corresponds to JavaScript `null`.
+///
+/// @description Checks if this value corresponds to JavaScript `null`.
+/// @author sgrekhov22@gmail.com
+
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+import '../../../Utils/expect.dart';
+import '../js_utils.dart';
+
+main() {
+  eval(r'''
+    var n = null;
+    var u = undefined;
+    var s = "s";
+  ''');
+
+  Expect.isTrue(globalContext["n"].isNull);
+  Expect.isFalse(42.toJS.isNull);
+  Expect.isFalse(globalContext["u"].isNull);
```

Comment:
This is the most interesting case. I don't know whether or not `isNull` should see any difference between `null` and `undefined`.

The documentation says

> Currently, there is no way to distinguish between JavaScript `undefined`
> and JavaScript `null` when compiling to Wasm. Therefore, this getter
> should only be used for code that compiles to JavaScript and will throw
> when compiling to Wasm.

.. so there may be a need to treat compilation to Wasm and compilation to JS differently.

Reply:
Thanks for the cc, Erik! And thanks for adding tests, Sergey!

Yes, when compiling to Wasm, we're forced to convert to Wasm's null ref regardless of whether the JS value is `undefined` or `null` in order to maintain the same nullability semantics for interop across all compilers, so there's no way of distinguishing.

In some future, we may provide an opt-in annotation users to preserve `undefined` separately, but that doesn't exist yet.

So, both `isNull` and `isUndefined` need to be handled differently based on the platforms.